### PR TITLE
Moodle 4.5 compatibility.

### PR DESCRIPTION
**Pull request by lucaboesch**
This would add the Moodle Plugin CI runs for all versions up to Moodle 4.5, drop the PHP ≤ 7.3 ones, and adhere to modern coding style.

### CODE REVIEW

**Comment by adrian-sarmas** [comment]
Thank you @lucaboesch for contributing. I am wondering if it's really necessary to drop support for versions of the PHP ≤ 7.3. I'd like the plugin to support as many Moodle versions as possible. Other that that, it looks good. I'll check it out asap.

**Comment by lucaboesch** [comment]
Hi Adrian

Support is not dropped and the Plugin still works in those versions.
It is just that the automated tests can not be run on ubuntu-latest in 2024 any more.

Best,
Luca

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -157,16 +185,6 @@ jobs:
         if: ${{ always() }}
         run: moodle-plugin-ci phplint
 
-      - name: PHP Copy/Paste Detector
-        continue-on-error: true # This step will show errors but will not fail
-        if: ${{ always() }}
-        run: moodle-plugin-ci phpcpd
-
-      - name: PHP Mess Detector
```
Comment (adrian-sarmas) [inline .github/workflows/main.yml:165]: PHP Mess Detector continues to be supported in Moodle Plugin CI. Could you share the reasoning behind your decision to remove it, @lucaboesch?

Comment (lucaboesch) [reply .github/workflows/main.yml:165]: I must have confused it with phpcpd. Sorry.
It is back in.

**Comment by adrian-sarmas** [comment]
Thank you @lucaboesch for your contribution.

**Pull request merged**